tests/integration_test.go

```diff
@@ -0,0 +1,368 @@
+	require.Equal(t, tx.GasPrice(), big.NewInt(10))
+	require.Equal(t, tx.Gas(), uint64(10))
+
+	addrBytes, err := hex.DecodeString(strings.TrimPrefix(addr, "0x"))
+	require.NoError(t, err)
+
+	require.Equal(t, tx.To()[:], addrBytes)
+	require.Equal(t, tx.To().Hex(), addr)
+	require.Equal(t, *tx.Value(), *big.NewInt(10))
+	require.Equal(t, tx.Data(), []byte{})
+}
+
+func TestInitWallet(t *testing.T) {
+	wallet, account := initWallet(t, gethaccounts.DefaultBaseDerivationPath)
+	defer wallet.Close()
+
+	pk := hex.EncodeToString(crypto.FromECDSAPub(account.PublicKey))
+
+	require.Equal(t, account.Address.Hex(), "0xbcf6368dF2C2999893064aDe8C4a4b1b6d3C077B")
+	require.Equal(t, pk, "0x045f53cbc346997423fe843e2ee6d24fd7832211000a65975ba81d53c87ad1e5c863a5adb3cb919014903f13a68c9a4682b56ff5df3db888a2cbc3dc8fae1ec0fb")
+}
+
+func TestInvalidAccount(t *testing.T) {
+	wallet, account := initWallet(t, gethaccounts.DefaultBaseDerivationPath)
+	defer wallet.Close()
+
+	account.Address = common.HexToAddress("0x3535353535353535353535353535353535353535")
+
+	sendAddr := "0x3636363636363636363636363636363636363636"
+	tx, err := CreateTx(
+		3, sendAddr, 10, big.NewInt(10), big.NewInt(10), make([]byte, 0),
+	)
+	require.NoError(t, err)
+
+	_, err = wallet.SignTx(account, tx, big.NewInt(0))
+	require.Error(t, err, "Expected error on signing with invalid account")
+}
+
+// Test deriving an account with path "m/44'/60'/0'/0/1"
+func TestAlternateDerivation(t *testing.T) {
+	path, err := gethaccounts.ParseDerivationPath("m/44'/60'/0'/0/1")
+	require.NoError(t, err)
+
+	wallet, account := initWallet(t, path)
+	defer wallet.Close()
+
+	addr := crypto.PubkeyToAddress(*account.PublicKey)
+	pk := hex.EncodeToString(crypto.FromECDSAPub(account.PublicKey))
+	require.Equal(t, pk, "0x044a5236e77ab81e094d7c6cfeac06d2e93fec455d01c7f80e22c592a89b44acebe99c2450425a184e5382362d5c52f5d996f12e73ccfb7694227f31b501e36ed7")
+	require.Equal(t, account.Address.Hex(), addr.Hex())
+}
+
+func TestLedgerSignTx1(t *testing.T) {
+	wallet, account := initWallet(t, gethaccounts.DefaultBaseDerivationPath)
+	defer wallet.Close()
+
+	addr := "0x3535353535353535353535353535353535353535"
+
+	tx, err := CreateTx(
+		3, addr, 10, big.NewInt(10), big.NewInt(10), make([]byte, 0),
+	)
+	require.NoError(t, err)
+
+	sigBytes, err := wallet.SignTx(account, tx, big.NewInt(0))
+	require.NoError(t, err)
+
+	sigHex := hex.EncodeToString(sigBytes)
+
+	// Test against signature generated using ethers.js
+	require.Equal(t, sigHex, "f85d030a0a9435353535353535353535353535353535353535350a801ca02e0b1b0ed24cd450488eb783e6c64ab0f1d681641970aef062434513731e829ca0721e7b6feedc989a8b114f3f622d5a525095b893b8ce81059e682f7333be3508")
+}
+
+func TestLedgerSignTx2(t *testing.T) {
+	wallet, account := initWallet(t, gethaccounts.DefaultBaseDerivationPath)
+	defer wallet.Close()
+
+	addr := "0x4646464646464646464646464646464646464646"
+
+	tx, err := CreateTx(
+		8, addr, 50, big.NewInt(5), big.NewInt(70), []byte{4, 6, 8, 10},
+	)
```

Comment:
Nit: refactor the `TestLedgerSignTxN` tests as [table driven tests](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests)

Reply:
@Vvaradinov and @4rgon4ut are taking care of this in their PR (https://github.com/evmos/evmos-ledger-go/pull/10)